src/libraries/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/Http2StreamWindowManager.cs

```diff
@@ -0,0 +1,257 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Diagnostics;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace System.Net.Http
+{
+    internal sealed partial class Http2Connection
+    {
+        private struct Http2StreamWindowManager
+        {
+            public const int StreamWindowUpdateRatio = 8;
+            private static readonly double StopWatchToTimesSpan = TimeSpan.TicksPerSecond / (double)Stopwatch.Frequency;
+
+            private readonly Http2Connection _connection;
+            private readonly Http2Stream _stream;
+
+            private readonly double _windowScaleThresholdMultiplier;
```

Comment:
Aren't `_windowScaleThresholdMultiplier` and `_maxStreamWindowSize` process wide constants? Could they be made static?

Reply:
Depends on the decision re discussion starting in https://github.com/dotnet/runtime/pull/54755#discussion_r660067631.

Reply:
Per discussion just now, we decided to keep these. But I agree, they should be static.